pkg/resources/envoy/poddisruptionbudget.go

```diff
@@ -0,0 +1,81 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package envoy
+
+import (
+	"fmt"
+
+	"github.com/banzaicloud/kafka-operator/api/v1beta1"
+	"github.com/banzaicloud/kafka-operator/pkg/resources/templates"
+	"github.com/banzaicloud/kafka-operator/pkg/util"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Reconciler) podDisruptionBudget(log logr.Logger, extListener v1beta1.ExternalListenerConfig,
+	ingressConfig v1beta1.IngressConfig, ingressConfigName, defaultIngressConfigName string) runtime.Object {
+	eListenerLabelName := util.ConstructEListenerLabelName(ingressConfigName, extListener.Name)
+
+	var deploymentName string
+	if ingressConfigName == util.IngressConfigGlobalName {
```

Comment:
Let's factor out the deployment name derivation logic into a separate function and use that here and on `envoy/envoy.go` as well.

Reply:
Took the liberty to make a separate function in utils for generating the Resource Names for Service, Deployment, ConfigMap, and Volume, and reused it for all of the above, as you mentioned in the comment above.